SignalServiceKit/src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -184,8 +184,7 @@ - (void)setExpirationForMessage:(TSMessage *)message
 
     // Don't clobber if multiple actions simultaneously triggered expiration.
     if (message.expireStartedAt == 0 || message.expireStartedAt > expirationStartedAt) {
-        message.expireStartedAt = expirationStartedAt;
-        [message saveWithTransaction:transaction];
+        [message updateWithExpireStartedAt:expirationStartedAt transaction:transaction];
```

Comment:
I also noticed that we weren't applying the `updateWith:...` pattern here.  Fixed.